src/components/boolean-search/boolean-table.tsx

```diff
@@ -0,0 +1,307 @@
+      { Header: 'With Pattern', accessor: 'pattern' },
+      { Header: 'At Location', accessor: 'location' },
+      {
+        Header: 'Actions',
+        accessor: 'actions',
+        Cell: ({ row }: any) => (
+          <div className='actions-button-wrapper'>
+            <button
+              onClick={(e) => handleClearFilter(e, row.index)}
+              className='chaise-btn chaise-btn-primary action-button'
+              title='Click to remove this filter'
+            >
+              <span className='fa fa-close'></span>
+            </button>
+            <button
+              onClick={(e) => handleRemoveRow(e, row.index)}
+              className={`chaise-btn chaise-btn-primary action-button ${rows.length === 1 ? 'disable-trash' : ''}`}
+              title='Click to remove this filter'
+            >
+              <span className='fa fa-trash'></span>
+            </button>
+          </div>
+        )
+      }
+    ],
+    [rows]
+  );
+
+  const data: any = useMemo(
+    () =>
+      rows?.map((row: any, index: any) => ({
+        // Map your rows data here
+        and: index === 0 ? '' : 'AND',
+        strength: (
+          <select
+            value={row.strength || defaultValues.strength}
+            onChange={(e) => handleChange(index, 'strength', e.target.value)}
+          >
+            {strength?.map((option: any, index: any) => {
+              return <option key={index}>{option.Strength}</option>;
+            })}
+          </select>
+        ),
+        source: (
+          <>
+            <input
+              className='red-text'
+              value={row.source.name || ''}
+              disabled
+              required
+              onChange={(e) => handleChange(index, 'source', e.target.value)}
+            />
+            {row.sourceInvalid ? <div className='red-text'>Please enter a value for this field.</div> : <></>}
+          </>
+        ),
+        stages: (
+          <>
+            <span>From: </span>
+            <select
+              className='stageDropdown'
+              value={row.stageFrom.Name || defaultValues.stageFrom.Name}
+              onChange={(e) => handleChange(index, 'stageFrom', e.target.value)}
+            >
+              {stageFrom?.map((option: any, index: any) => {
+                return <option key={index}>{option.Name}</option>;
+              })}
+            </select>
+            <span>To: </span>
+            <select
+              className='stageDropdown'
+              value={row.stageTo.Name || defaultValues.stageTo.Name}
+              onChange={(e) => handleChange(index, 'stageTo', e.target.value)}
+            >
+              {row.toStageOptions?.map((option: any, index: any) => {
+                return <option key={index}>{option.Name}</option>;
+              })}
+            </select>
+          </>
+        ),
+        pattern: (
+          <select
```

Comment:
There should be an option in the select to set the pattern to empty string (which clears it from the query).